docs/dg/dev/third-party-module-development/ensure-quality.md

```diff
@@ -0,0 +1,97 @@
+  - title: Go to the Next Step.
+    link: docs/dg/dev/third-party-module-development/publish-code.html
+---
+
+<style>
+  table th, table td {
+    white-space: nowrap;
+  }
+</style>
+
+**TBD: Commands must be updated after https://spryker.atlassian.net/browse/FRW-8436 is done (Check if we need it at all)  **
+
+## Tools
+
+To make sure that code quality matched the Spryker standards it is important to execute code quality tools, here is the list:
+
+| Command                                                                                   | Description                                                             |
+|:------------------------------------------------------------------------------------------|:------------------------------------------------------------------------| 
+| console code:propel:validate-abstract -m YourCompanyName.YourModuleName                   | Validates that propel entity/query classes are implemented properly.    |
+| console propel:schema:validate                                                            | Validates Propel the schema files.                                      |
+| console propel:schema:validate-xml-names                                                  | Validates Propel XML element name rules for schema files.               |
+| console transfer:validate                                                                 | Validates transfer XML definition files.                                |
+| spryker-dev-console dev:validate-module-schemas YourCompanyName.YourModuleName            | Validates Propel schema definitions.                                    |
+| spryker-dev-console dev:validate-module-databuilders YourCompanyName.YourModuleName       | Validates databuilder XML files.                                        |
+| spryker-dev-console dev:module-meta-files -d -v YourCompanyName.YourModuleName            | Validates that all the mandatory meta files exist.                      |
+| spryker-dev-console dev:composer:update-json-files -d -v YourCompanyName.YourModuleName   | Validates that all the necessary module dependencies are added.         |
+| console dev:composer:validate-json-files YourCompanyName.YourModuleName                   | Validates composer.json file to make sure that it's structure is valid. |
+| console code:sniff:style -m YourCompanyName.YourModuleName                                | Sniffs and fixes code style.                                            |
+| spryker-dev-console dev:validate-module-transfers -m YourCompanyName.YourModuleName       | Validates DTO definition for the module.                                |
+| console code:sniff:architecture -v -m YourCompanyName.YourModuleName                      | Validates module architecture to find a common architecture mistakes.   |
+| console code:phpstan -m YourCompanyName.YourModuleName                                    | Runs PHPStan static analyzer to ensure the code quality.                |
```

Comment:
Should these commands be executed as part of module's own CI? Because the usage of these commands implies quite a few external dependencies that have to be listed in the `require-dev` sections of `composer.json` files.

Reply:
Please ignore this comment for now, it is under discussion with @eduard-melnytskyi and @vlunov-spryker